spectec/spec/wasm-2.0/1-syntax.watsup

```diff
@@ -296,8 +296,8 @@ syntax vbinop_(Jnn X N) =
   | ADD_SAT sx hint(show ADD_SAT#_#%)      -- if $sizemm(Jnn) <= `16
   | SUB_SAT sx hint(show SUB_SAT#_#%)      -- if $sizemm(Jnn) <= `16
   | MUL                                    -- if $sizemm(Jnn) >= `16
-  | AVGR_U                                 -- if $sizemm(Jnn) <= `16
-  | Q15MULR_SAT_S                          -- if $sizemm(Jnn) = `16
+  | AVGR U                                 -- if $sizemm(Jnn) <= `16
```

Comment:
So, here was the problem:

It was expected that `AVGR` (or `Q15MULR_SAT`) is parsed as:
`CaseE (AVGR, [CaseE (U, [])])`,
but instead, parsed as a `CaseE` with zero arguments:
`CaseE (AVGR U, [])`.
i.e., `U` is considered as a part of the `vbinop`'s name itself, not as a separated `sx`.

This leads to the mismatch with the manually constructed CaseV from construct.ml:1415:
`  | V128Op.AvgrU -> Some (caseV ("AVGR", [nullary "U"]))`,
resulting in the fall-through to the execution of PMAX as described before.

Perhaps something like
```
| AVGR sx        -- if $sizemm(Jnn) <= `16 -- if sx = U
```
would solve the issue.

Reply:
Ah, I see! Thanks @f52985 for figuring that out. I see that translate.ml simply drops all but the first atom from a mixop. So wouldn't it work if I just tweaked construct.ml to translate to/from `nullary "AVGR"`?